arch/arc/core/irq_manage.c

```diff
@@ -95,10 +95,10 @@ void z_arc_firq_stack_set(void)
 
 void arch_irq_enable(unsigned int irq)
 {
-	unsigned int key = irq_lock();
+	unsigned int key = arch_irq_lock();
```

Comment:
I'd move `arch_irq_lock()` from here right to the `z_arc_v2_irq_unit_irq_enable_set()` so it's clear why do we need it.

See what we have in `z_arc_v2_irq_unit_irq_enable_set()`
```c
void z_arc_v2_irq_unit_irq_enable_set(
	int irq,
	unsigned char enable
	)
{
	z_arc_v2_aux_reg_write(_ARC_V2_IRQ_SELECT, irq);
	z_arc_v2_aux_reg_write(_ARC_V2_IRQ_SELECT, enable);
}
```

Obviously we don't want to re-enter it in between setting `_ARC_V2_IRQ_SELECT` and writing to `_ARC_V2_IRQ_SELECT`. Because in that case we'll discard the first IRQ we meant to select and simply enable "the second" IRQ twice.

Reply:
There's a warning and rationale given in arcv2_irq_unit.h that disabling interrupts is responsibility of the caller. Rationale is performance in case of multiple calls to IRQ handling - don't know if that's happening in current code, or a more defensive programming like Alexey proposes and always lock in the low-level irq control fuctions makes more sense...

Reply:
make sense, i will move it to z_arc_v2_irq_unit_irq_enable_set